devito/passes/iet/parpragma.py

```diff
@@ -274,13 +273,6 @@ def _make_parregion(self, partree, parrays):
         return self.Region(partree)
 
     def _make_guard(self, parregion):
```

Comment:
if _make_guard becomes completely useless, both here and in all subclasses, then we can drop it entirely?

Reply:
It is used in device operators, right?
```
        # SparseFunctions may occasionally degenerate to zero-size arrays. In such
        # a case, a copy-in produces a `nil` pointer on the device. To fire up a
        # parallel loop we must ensure none of the SparseFunction pointers are `nil`
```